src/main/groovy/com/vanniktech/android/apk/size/ApkSizeTask.groovy

```diff
@@ -33,9 +35,20 @@ class ApkSizeTask extends DefaultTask {
                 appendableStream.println(bytes + "," + kiloBytes + "," + megaBytes)
             }
         }
+
+        failBuildMaxMethods(apkSize, extension.maxApkSize)
```

Comment:
Run this method last, so the user(developer) can see the exception after the reports have printed to STDOUT and saved to disk.